pkg/skaffold/test/test_factory.go

```diff
@@ -113,42 +113,56 @@ func (t FullTester) Test(ctx context.Context, out io.Writer, bRes []build.Artifa
 
 func (t FullTester) runTests(ctx context.Context, out io.Writer, bRes []build.Artifact) error {
 	for _, test := range t.testCases {
-		if err := t.runStructureTests(ctx, out, bRes, test); err != nil {
-			return fmt.Errorf("running structure tests: %w", err)
+		fqn, found := resolveArtifactImageTag(test.ImageName, bRes)
+		if !found {
+			logrus.Debugln("Skipping tests for", test.ImageName, "since it wasn't built")
+			return nil
 		}
-	}
-
-	return nil
-}
 
-func (t FullTester) runStructureTests(ctx context.Context, out io.Writer, bRes []build.Artifact, tc *latest.TestCase) error {
-	if len(tc.StructureTests) == 0 {
-		return nil
-	}
-
-	fqn, found := resolveArtifactImageTag(tc.ImageName, bRes)
-	if !found {
-		logrus.Debugln("Skipping tests for", tc.ImageName, "since it wasn't built")
-		return nil
-	}
+		if imageIsLocal, err := t.imagesAreLocal(test.ImageName); err != nil {
```

Comment:
This is specific to container structure test. 
Can we move this code to structure.Test?

Reply:
One of the scenarios for custom tests is to run validation tests on the image before deploying. I can move it to a util.

Reply:
that is great point. However, its the drawback of container structure test, that it cannot run validation tests on a *remote* image. 
Most tools, can pull images and run validation test on them.

Reply:
Moved image download logic to `structure.Test`.